### PR TITLE
add _augment_coords implementation for many ImageAugmentor subclasses

### PR DESCRIPTION
**Pull request by haamoon**
I'll do the rest later. I have attached the test code (requires matplotlib and skimage). Let me know what you think. I'll add AugmentImageCoordinates shortly.

[test.zip](https://github.com/ppwwyyxx/tensorpack/files/1154688/test.zip)

### CODE REVIEW

File: tensorpack/dataflow/imgaug/crop.py [hunk 1]
```diff
@@ -127,8 +137,10 @@ def _get_augment_params(self, img):
     def _augment(self, img, newbox):
         return newbox.roi(img)
 
-    def _augment_coords(self, coords, param):
-        raise NotImplementedError()
+    def _augment_coords(self, coords, newbox):
```
Comment (ppwwyyxx) [inline tensorpack/dataflow/imgaug/crop.py]: Don't have to worry too much about these weird augmentors.. They are not quite well defined and not used by anyone, I think.

File: tensorpack/dataflow/imgaug/noname.py [hunk 2]
```diff
@@ -70,6 +83,13 @@ def _augment(self, img, _):
             ret = ret[:, :, np.newaxis]
         return ret
 
+    def _augment_coords(self, coords, param):
+        h, w = param
+        #TODO Can self.interp value change these computation?
```
Comment (ppwwyyxx) [inline tensorpack/dataflow/imgaug/noname.py]: I think interpolation doesn't affect the alignment in opencv.

File: tensorpack/dataflow/imgaug/crop.py (hunk 1 above)
Comment (haamoon) [inline tensorpack/dataflow/imgaug/crop.py]: I might need this in faster rcnn :)

File: tensorpack/dataflow/imgaug/noname.py (hunk 2 above)
Comment (haamoon) [inline tensorpack/dataflow/imgaug/noname.py]: Just deleted corresponding TODOs.

**Comment by haamoon** [comment]
Just tested AugmentImageCoordinates. Use attached code to test all.
[Test.zip](https://github.com/ppwwyyxx/tensorpack/files/1154887/Test.zip)

File: tensorpack/dataflow/imgaug/crop.py (hunk 1 above)
Comment (ppwwyyxx) [inline tensorpack/dataflow/imgaug/crop.py]: Some sort of random bounding box perturbation is definitely helpful, but the problem is there isn't a standard way of doing this. If something isn't a standard, I think it's best to let users write their own version of it than to provide some arbitrary implementation. I might just mark this augmentor as deprecated.

**Comment by haamoon** [comment]
Yes, that makes sense.

-------- Original message --------

File: tensorpack/dataflow/imgaug/crop.py (hunk 1 above)
Comment (haamoon) [inline tensorpack/dataflow/imgaug/crop.py]: Yes, that makes sense.

**Comment by ppwwyyxx** [comment]
Thank you very much!

**Comment by haamoon** [comment]
Absolutely. Let me know if you need help on other stuff. I would like to collaborate with you.

**Pull request merged**